app/views/helpers/ViewDates.scala

```diff
@@ -16,28 +16,74 @@
 
 package views.helpers
 
-import java.time.ZoneId
+import play.api.Logging
+import play.api.i18n.Messages
+
 import java.time.format.DateTimeFormatter
+import java.time.temporal.ChronoField.MONTH_OF_YEAR
 import java.time.temporal.TemporalAccessor
+import java.time.{Instant, LocalDate, Month, ZoneId}
+import scala.util.{Failure, Success, Try}
+
+object ViewDates extends Logging {
+
+  def formatDate(temporal: TemporalAccessor)(implicit messages: Messages): String =
+    translateMonthOnWelsh(temporal, dateFormatter.format(temporal))
+
+  def formatDateAtTime(temporal: TemporalAccessor)(implicit messages: Messages): String = {
+    val result = translateAmPmOnWelsh(dateTimeFormatter.format(temporal))
+    translateMonthOnWelsh(temporal, result)
+  }
 
-object ViewDates {
+  def formatTimeDate(temporal: TemporalAccessor)(implicit messages: Messages): String =
+    translateMonthOnWelsh(temporal, timeDateFormatter.format(temporal))
 
-  def formatDate(temporal: TemporalAccessor): String = dateFormatter.format(temporal)
+  private def translateAmPmOnWelsh(result: String)(implicit messages: Messages): String =
```

Comment:
Can we rename `translateAmPmOnWelsh` to `translateAmPmToWelsh` (reads a bit funny with 'On')